xds/internal/resolver/serviceconfig.go

```diff
@@ -83,12 +86,23 @@ func (r *xdsResolver) pruneActiveClusters() {
 // serviceConfigJSON produces a service config in JSON format representing all
 // the clusters referenced in activeClusters.  This includes clusters with zero
 // references, so they must be pruned first.
-func serviceConfigJSON(activeClusters map[string]*clusterInfo) ([]byte, error) {
+func serviceConfigJSON(activeClusters map[string]*clusterInfo, clusterSpecifierPlugins map[string]clusterspecifier.BalancerConfig) ([]byte, error) {
 	// Generate children (all entries in activeClusters).
 	children := make(map[string]xdsChildConfig)
 	for cluster := range activeClusters {
-		children[cluster] = xdsChildConfig{
-			ChildPolicy: newBalancerConfig(cdsName, cdsBalancerConfig{Cluster: cluster}),
+		// Look into cluster specifier plugins, which hasn't had any prefix attached to it's cluster specifier plugin names,
+		// to determine the LB Config if the cluster is a CSP.
+		cspCfg, ok := clusterSpecifierPlugins[strings.TrimPrefix(cluster, clusterSpecifierPluginPrefix)]
```

Comment:
So... _if_ a CSP is called "cluster:foo" _and_ a cluster named "foo" exists, then "cluster:foo" will be found in the CSP map, even though that isn't what was intended.

Should we be looking at whether the `cluster` from `activeClusters`'s key `StartsWith` `clusterSpecifierPluginPrefix` instead here?

Reply:
I agree that there is a correctness issue here, but I disagree that this issue can be fixed by what you are describing. What about if a cluster has the prefix "cluster_specifier_plugin:foo", starts with would return true, and could have a configuration in clusterSpecifierPlugins keyed on foo.

Reply:
Simply persisted the config in clusterInfo, which solves this correctness issue.